saleor/graphql/product/bulk_mutations/product_variant_bulk_update.py

```diff
@@ -191,74 +477,126 @@ def update_variants(cls, info, cleaned_inputs_map, index_error_map):
+        if stocks_data := stocks_input.get("create"):
+            stocks_to_create += [
+                warehouse_models.Stock(
+                    product_variant=variant,
+                    warehouse=stock_data["warehouse"],
+                    quantity=stock_data["quantity"],
                 )
+                for stock_data in stocks_data
+            ]
+        if stocks_data := stocks_input.get("update"):
+            stocks_to_update += [stock_data["stock"] for stock_data in stocks_data]
 
-                if is_created or (stock.quantity <= 0 and stock_data["quantity"] > 0):
-                    transaction.on_commit(
-                        lambda: manager.product_variant_back_in_stock(stock)
-                    )
+    @classmethod
+    def prepare_channel_listings(
+        cls, variant, listings_input, listings_to_create, listings_to_update
+    ):
+        if listings_data := listings_input.get("create"):
+            listings_to_create += [
+                models.ProductVariantChannelListing(
+                    channel=listing_data["channel"],
+                    variant=variant,
+                    price_amount=listing_data["price"],
+                    cost_price_amount=listing_data.get("cost_price"),
+                    currency=listing_data["channel"].currency_code,
+                    preorder_quantity_threshold=listing_data.get("preorder_threshold"),
+                )
+                for listing_data in listings_data
+            ]
 
-                if stock_data["quantity"] <= 0:
-                    transaction.on_commit(
-                        lambda: manager.product_variant_out_of_stock(stock)
-                    )
+        if listings_data := listings_input.get("update"):
+            for listing_data in listings_data:
+                listing = listing_data["channel_listings"]
+                if "preorder_threshold" in listing_data:
+                    listing.preorder_quantity_threshold = listing_data[
+                        "preorder_threshold"
+                    ]
+                if "price" in listing_data:
+                    listing.price_amount = listing_data["price"]
+                if "cost_price" in listing_data:
+                    listing.cost_price_amount = listing_data["cost_price"]
+                listings_to_update.append(listing)
 
-                stock.quantity = stock_data["quantity"]
-                stocks.append(stock)
+    @classmethod
+    @traced_atomic_transaction()
+    def save_variants(cls, variants_data_with_errors_list):
+        variants_to_update: list = []
+        stocks_to_create: list = []
+        stocks_to_update: list = []
+        stocks_to_remove: list = []
+        listings_to_create: list = []
+        listings_to_update: list = []
+        listings_to_remove: list = []
+
+        # prepare instances
+        for variant_data in variants_data_with_errors_list:
+            variant = variant_data["instance"]
+            if variant:
```

Comment:
It is a cosmetic thing, but if you reverse the condition and use `continue` here, you will reduce one indentation and IMHO improve the readability